### PR TITLE
Add note on generic client alternative

### PR DESCRIPTION
**Pull request by MattKuhr**
## What Has Changed?

Adding an info about the generic client under the typed API docu + troubleshooting section.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 1]
```diff
@@ -25,6 +25,14 @@ The typed OData v2 client allows to build type-safe OData v2 requests for a give
 The Java classes represent the _data model_ and the available _operations_ of the service.
 As a consequence all requests that are built through the typed OData v2 client are not only _syntactically valid_ but also _semantically valid_.
 
+:::info
```
Comment (artemkovalyov) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:28]:
```suggestion
:::info Generic OData Client
```

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 2]
```diff
@@ -25,6 +25,14 @@ The typed OData v2 client allows to build type-safe OData v2 requests for a give
...
+The type safe client may in some situations not be suitable for your needs.
```
Comment (artemkovalyov) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:29]:
```suggestion
The type-safe client by the SAP Cloud SDK may in some situations not be suitable for your needs.
```

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 3]
```diff
@@ -25,6 +25,14 @@ The typed OData v2 client allows to build type-safe OData v2 requests for a give
...
+For instance a service might not fully conform to the OData standard, forcing you to perform customized requests.
+Or the typed API doesn't support a specific feature just yet.
```
Comment (artemkovalyov) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:31]:
```suggestion
Or the typed API doesn't support a specific OData feature just yet.
```

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 4]
```diff
@@ -25,6 +25,14 @@ The typed OData v2 client allows to build type-safe OData v2 requests for a give
...
+
+For such cases you can leverage the more [low-level generic OData Client](./generic-untyped-odata-client).
```
Comment (artemkovalyov) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:33]:
```suggestion
For such cases, we recommend leveraging our [low-level generic OData Client](./generic-untyped-odata-client) or fallback directly to HTTP Client level if that's not enough.
In all the other situations typed client is preferable and considered the best practice. 
```

**Review by artemkovalyov — APPROVED** [review]
I proposed some improvements. No hard feelings to leave it the submitted from either.

**Comment by artemkovalyov** [comment]
You can apply what you like to the second note as well. Thanks for taking care.

**Comment by MattKuhr** [comment]
Great suggestions, I have incorporated them

**Pull request merged**